runtime-light/stdlib/fork/wait-queue.h

```diff
@@ -0,0 +1,96 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2024 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#pragma once
+
+#include <algorithm>
+#include <coroutine>
+#include <cstdint>
+
+#include "runtime-core/memory-resource/resource_allocator.h"
+#include "runtime-core/memory-resource/unsynchronized_pool_resource.h"
+#include "runtime-light/coroutine/task.h"
+#include "runtime-light/stdlib/fork/fork.h"
+#include "runtime-light/utils/concepts.h"
+
+class WaitQueue {
```

Comment:
It looks quite strange. We should not work with raw coroutine handlers unless we develop and awaitable. In my opinion, `WaitQueue` can be completely removed. `WaitQueueContext` can be simplified a lot. Also, keep in mind that wait queues can be used to wait for job workers

Reply:
merge `wait_next_t` and `WaitQueue`